apptests/test_testrunners.py

```diff
@@ -0,0 +1,234 @@
+        mock_video = Mock()
+        mock_open.return_value = mock_file
+
+        mock_from_dict.return_value = mock_video
+
+        self.runner.tester.transcriber = None
+        self.runner.tester.normalizer = None
+        self.runner._iterations = 2
+
+        self.runner._testplan_path = "mock_testplan_path"
+
+        with patch("builtins.open"), patch(
+            "src.dataclasses.youtube_video.YouTubeVideo.from_dict"
+        ), patch("youtube_runner.YouTubeTestRunner.save_results"), patch(
+            "generators.youtube_generator.generate"
+        ):
+            with self.assertRaises(ValueError) as context:
+                self.runner.run()
+
+        self.assertEqual(str(context.exception), "Transcriber is None")
+
+
+details_result = [
+    {
+        "channelId": "UCWlvkaA27BCrcYG6gA0K8UA",
+        "channelTitle": "SMS Frankfurt Group Travel",
+        "defaultAudioLanguage": "en",
+        "duration": "PT14M41S",
+        "generatedTranscripts": ["en"],
+        "manuallyCreatedTranscripts": ["en", "de"],
+        "publishTime": "2016-12-21T11:09:18Z",
+        "title": "We create your Bucket List Trips and Events | Discover the Baltics with SMS Frankfurt Group Travel",
+        "videoId": "YazZwd48ws0",
+    }
+]
+
+search_request = {
+    "args": {
+        "maxResults": 10,
+        "pageToken": None,
+        "q": "Travel & Events",
+        "regionCode": "US",
+        "relevanceLanguage": "en",
+        "topicId": None,
+        "videoCategoryId": "19",
+        "videoDuration": "medium",
+        "videoLicense": "creativeCommon",
+    }
+}
+
+testplan = deepcopy(search_request)
+testplan["items"] = [
+    {
+        "channelId": "UCWlvkaA27BCrcYG6gA0K8UA",
+        "channelTitle": "SMS Frankfurt Group Travel",
+        "defaultAudioLanguage": "en",
+        "duration": "PT14M41S",
+        "manuallyCreatedTranscripts": ["en"],
+        "publishTime": "2016-12-21T11:09:18Z",
+        "title": "We create your Bucket List Trips and Events | Discover the Baltics with SMS Frankfurt Group Travel",
+        "videoId": "YazZwd48ws0",
+    },
+    {
+        "channelId": "UCmqdOJinYxakne7lyAXY5zw",
+        "channelTitle": "Passport Kings Travel",
+        "defaultAudioLanguage": "en",
+        "duration": "PT6M30S",
+        "generatedTranscripts": [],
+        "manuallyCreatedTranscripts": ["en"],
+        "publishTime": "2015-04-08T02:34:15Z",
+        "title": "Men Just Traveling Abroad for Prostitutes? Passport Kings Travel Video",
+        "videoId": "gI8VlFK5Jp4",
+    },
+    {
+        "channelId": "UCkeTA80xeSa9POCGRpWTIfQ",
+        "channelTitle": "travelingisrael.com",
+        "defaultAudioLanguage": "en",
+        "duration": "PT16M18S",
+        "manuallyCreatedTranscripts": ["en"],
+        "publishTime": "2023-10-18T12:38:57Z",
+        "title": "Why are there so many Palestinian casualties? (The Israeli perspective) sub: DE, ES, FR, IT",
+        "videoId": "OF95GenB1JI",
+    },
+]
```

Comment:
Variables are not used or can be replaced by loading the `simpleTest.json` file.